src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -7652,5 +7652,140 @@ public void RecordDeclaration()
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+            Verify(
+                symbol.ToDisplayParts(format),
+                "Object? P",
+                SymbolDisplayPartKind.ClassName,
+                SymbolDisplayPartKind.Punctuation,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [Fact]
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        public void NullableAttributes_MayBeNullOnClassProperty_WithoutNullableModifier()
+        {
+            var source =
+@"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+public class A<T>
+{
+    [MaybeNull]
+    public T P { get; }
+}
+
+public class B
+{
+    void M()
+    {
+        var a = new A<object>();
+        _ = a.P;
+    }
+}
+";
+            var compilation = CreateCompilation(new[] { source, MaybeNullAttributeDefinition });
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            // look for memberAccess "a.P"
+            var memberAccess = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().Single().Right;
+            var symbol = model.GetSymbolInfo(memberAccess).Symbol;
+            var format = new SymbolDisplayFormat(
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeType | SymbolDisplayMemberOptions.IncludeModifiers,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.None); // without SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier
+
+            Verify(
+                symbol.ToDisplayParts(format),
+                "Object P",
+                SymbolDisplayPartKind.ClassName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [Fact]
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        public void NullableAttributes_MayBeNullOnStructProperty()
+        {
+            var source =
+@"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+public class A<T>
+{
+    [MaybeNull]
+    public T P { get; }
+}
+
+public class B
+{
+    void M()
+    {
+        var a = new A<int>();
+        _ = a.P;
+    }
+}
+";
+            var compilation = CreateCompilation(new[] { source, MaybeNullAttributeDefinition });
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            var memberAccess = tree.GetRoot().DescendantNodes().OfType<AssignmentExpressionSyntax>().Single().Right;
+            var symbol = model.GetSymbolInfo(memberAccess).Symbol;
```

Comment:
>var symbol = model.GetSymbolInfo(memberAccess).Symbol; [](start = 12, length = 54)

For this scenario, I would also avoid using semantic model. There is an API to ```Construct``` a generic type. There is also a trick to get constructed type by using it as base type of some other type, etc.
For example, define class like:
```
class C : A<int> {}
```
Locate the type, get its base type, get member from the base type: ```compilation.GetTypeByMetadataName("C").BaseType.GetMembers("P").Single()```

Reply:
That pattern (compilation -> tree -> model -> GetSymbolInfo) is used numerous times in that test class before. I just followed that pattern. With this pattern, it is also clear that I ask for `P` on a closed generic type.

Reply:
> That pattern (compilation -> tree -> model -> GetSymbolInfo) is used numerous times in that test class before. I just followed that pattern. With this pattern, it is also clear that I ask for P on a closed generic type.

I assume that whoever wrote those tests didn't know any better. Usage of  SemanticModel when not necessary creates confusion and also an unnecessary overhead during execution.

---
In reply to: [558516077](https://github.com/dotnet/roslyn/pull/50527#discussion_r558516077) [](ancestors = 558516077)

Reply:
Done in 71b3eac7351c87d232db82d3a8ed5c62b1f1e29e